[DOCS] Reorganize README (#900)
elastic/elasticsearch-ruby: docs/examples/README.md

# Doc Examples

In this directory you can find:

> - `apm`: A practical example of integrating elasticsearch-ruby and [Elastic APM](https://www.elastic.co/apm) with the [Ruby APM Agent](https://github.com/elastic/apm-agent-ruby). Check [the README](https://github.com/elastic/elasticsearch-ruby/tree/master/docs/examples/apm/README.md#observability-example) for more information.

Reviewer: Nit. This is fine as-is.
```suggestion
- `apm`: A practical example of integrating elasticsearch-ruby and [Elastic APM](https://www.elastic.co/apm) with the [Ruby APM Agent](https://github.com/elastic/apm-agent-ruby). See the [the README](https://github.com/elastic/elasticsearch-ruby/tree/master/docs/examples/apm/README.md#observability-example).
```

> - `guide`: Contains the generated Ruby code examples for the [Elasticsearch Reference](https://www.elastic.co/guide/en/elasticsearch/reference/master/index.html).

Reviewer: We use the `current` version in the root README.md. I'd use that here for consistency.
```suggestion
- `guide`: Contains the generated Ruby code examples for the [Elasticsearch Reference](https://www.elastic.co/guide/en/elasticsearch/reference/current/index.html).
```

> - `percolator`: An example of using the percolator with Elasticsearch 5.x and higher.

Reviewer: ```suggestion
- `percolator`: An example using a percolator in Elasticsearch version 5.x and higher.
```

> - `rabbitmq`: Example of listening to a RabbitMQ queue and indexing the payload.

Reviewer: I reworded here, but feel free to ignore if it's somehow changed the meaning.
```suggestion
- `rabbitmq`: Example of indexing the payload of a RabbitMQ queue.
```
